table/tables/tables.go

```diff
@@ -353,6 +396,41 @@ func (t *Table) getRollbackableMemStore(ctx sessionctx.Context) kv.RetrieverMuta
 	return bs
 }
 
+// locatePartition returns the partition ID of the input record.
+func (t *Table) locatePartition(ctx sessionctx.Context, r []types.Datum) (int64, error) {
```

Comment:
Can we avoid the usage of `[]types.Datum`, it should be deprecated in the near future. `chunk.Row` is preferred.

Reply:
This file highly depends on `[]types.Datum`, the caller provide `[]types.Datum` but not `types.Row`.
I know they're the same, but we can refine the code when `[]types.Datum` is removed.

Reply:
I think we can convert `[]types.Datum` to `chunk.Row` before we calling this function to reduce the future work when we removing `[]types.Datum`.

Reply:
Although coprocessor is chunk-based, it's insert here, and insert is still row-based.
In the OLTP mode, change the representation may not bring us some benefits.
Further more,  if we only change the parameter to `types.Row` here, it involves type converting,  result in a small performance penalty.
@zz-jason

Reply:
@zz-jason 
I think we do it in the future is more efficient, one PR replace many of the datums.

Reply:
What's the benefit ?

Reply:
@zz-jason 
The conversion is redundant work, it has both performance and complexity cost, if the caller pass `chunk.Row` directly, we don't need to do the conversion and pay the cost.

Reply:
@coocood For now I think we should consider the correctness, not the performance